conda_build/api.py

```diff
@@ -108,20 +114,28 @@ def render(
     return list(output_metas.values())
 
 
-def output_yaml(metadata, file_path=None, suppress_outputs=False):
+def output_yaml(
+    metadata: MetaData,
+    file_path: str | os.PathLike | Path | None = None,
```

Comment:
If you prefer, there's:
```suggestion
    file_path: Optional[str | os.PathLike | Path] = None,
```
I've been using `Optional` for readability as that's the terminology many other languages use for `| None`.

Reply:
Doing so is counter to [UP007](https://docs.astral.sh/ruff/rules/non-pep604-annotation/) and will raise a linting error